gourmet/gtk_extras/dialog_extras.py

```diff
@@ -1060,23 +1060,10 @@ def run (self):
         """Run our dialog and return the filename or None"""
         response = self.fsd.run()
         if response == Gtk.ResponseType.OK:
-            # Gtk.FileChooser by default returns a UTF-8 encoded string, see
-            # http://www.pyGtk.org/pygtk2reference/class-gtkfilechooser.html#FileNamesAndEncodings
-            # Functions like Pillow's Image.open() (which is used for adding
-            # an image to a recipe) however require a unicode string.
-            # We need to do this as users wouldn't be able to e.g. add images with
-            # diacritics in their paths.
-
             if self.multiple:
-                #if vfs_available:
-                #    fn = self.fsd.get_uris()
-                #else:
-                fn = [str(f, 'utf-8') for f in self.fsd.get_filenames()]
+                fn = self.fsd.get_filenames()
```

Comment:
Is it okay that it used to be a _list_ of something (no idea what, filenames?) and now it's just _something_?

Reply:
We return it and it probably should be a list of some caller wants to sort it. Can we have type annotations for this function?

Reply:
Yes: so it's possible to select several filenames in certain cases, where that function then returns a list of string filenames. Otherwise, a single string is returned, which the user indicated in the `FileChooserDialog`, unless the user cancels, where `None` is returned.